Samples/NetworkMediaStreamer/RTPMediaStreamer/src/RTPStreamSink.cpp

```diff
@@ -0,0 +1,433 @@
+    BYTE* pOut = m_pTxBuf.data();
+    size_t szOut = m_MTUSize;
+    BYTE* pOutEnd = pOut + szOut;
+    BYTE* sc1 = sc;
+    auto pOutCurr = pOut + rtpHeaderSize + stapAtypeHeaderSize;
+    uint32_t numNalsToSend = 0;
+    while (sc < (bufIn + szIn))
+    {
+        auto sc0 = sc;
+        while ((sc < bufIn + szIn) && !(*sc++));
+        sc1 = FindSC(sc, bufIn + szIn);
+        size_t nalsz = sc1 - sc;
+        if (pOutCurr + nalsz + stapASzHeaderSize > pOutEnd)
+        {
+            if (numNalsToSend == 0)
+            {
+                // STAP-A FU
+                pOut[rtpHeaderSize] = 28; // STAP-A FU type
+                pOut[rtpHeaderSize] |= (sc[0] & 0x60);
+                pOut[rtpHeaderSize + 1] = sc[0] & 0x1F;
+                pOut[rtpHeaderSize + 1] |= (1 << 7);
+                sc++; nalsz--;
+                auto maxSz = szOut - (rtpHeaderSize + stapAtypeHeaderSize + 1);
+                while (nalsz > maxSz)
+                {
+                    memcpy(&pOut[rtpHeaderSize + stapAtypeHeaderSize + 1], sc, maxSz);
+                    m_pTxBuf.Length((uint32_t)szOut);
+                    SendPacket(m_pTxBuf, llSampleTime, false);
+                    nalsz -= maxSz;
+                    sc += maxSz;
+                    pOut[rtpHeaderSize + 1] &= (~(1 << 7));
+                }
+
+                memcpy(&pOut[rtpHeaderSize + stapAtypeHeaderSize + 1], sc, nalsz);
+                pOut[rtpHeaderSize + 1] |= (1 << 6);
+                m_pTxBuf.Length(uint32_t(nalsz + rtpHeaderSize + stapAtypeHeaderSize + 1));
+                SendPacket(m_pTxBuf, llSampleTime, (sc1 == (bufIn + szIn)));
+                sc += nalsz;
+                sc = sc1;
+            }
+            //else if (numNalsToSend == 1)
+            //{
+            //    //if only this one nal fit into this packet, send as single non-stap nal
+            //    //PacketizeMode0(sc0, sc1 - sc0, llSampleTime);
+            //    
+            //}
+            else
+            {
+                pOut[rtpHeaderSize] = 24; //STAP-A type
+                m_pTxBuf.Length((uint32_t)(pOutCurr - pOut));
+                SendPacket(m_pTxBuf, llSampleTime, (sc1 == (bufIn + szIn)));
+                sc = sc0;
+            }
+            numNalsToSend = 0;
+            pOutCurr = pOut + rtpHeaderSize + stapAtypeHeaderSize;
+        }
+        else
+        {
+            pOutCurr[0] = (BYTE)((nalsz & 0xFF00) >> 8);
+            pOutCurr[1] = (BYTE)(nalsz & 0x00FF);
+            memcpy(&pOutCurr[stapASzHeaderSize], sc, nalsz);
+            pOutCurr += (nalsz + stapASzHeaderSize);
+            numNalsToSend++;
+            sc = sc1;
+        }
+
+    }
+
+    if (numNalsToSend)
+    {
+        pOut[rtpHeaderSize] = 24;
+        m_pTxBuf.Length((uint32_t)(pOutCurr - pOut));
+        SendPacket(m_pTxBuf, llSampleTime, true);
+    }
+
+}
+
+STDMETHODIMP RTPVideoStreamSink::AddTransportHandler(ABI::PacketHandler* packethandler, LPCWSTR protocol /*= L"rtp"*/, LPCWSTR params /*=L""*/)

```

Comment:
noexcepts on all the stdMethodimp

Reply:
all of these have Exception boundary. 
I am changing these to your suggested syntax of -
foo() try{ /* foo defn*/} catch()

---
In reply to: [491172222](https://github.com/microsoft/Windows-Camera/pull/13#discussion_r491172222) [](ancestors = 491172222)